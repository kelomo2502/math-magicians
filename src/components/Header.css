.links {
  display: flex;
  justify-content: flex-end;
}

.link-item {
  padding: 0 8px;
  border-left: 2px solid;
}

.first-item {
  border: 0;
}
