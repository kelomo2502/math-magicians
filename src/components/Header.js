import React from 'react';
import { Link as ReachRouterLink } from 'react-router-dom';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  border-bottom: solid #e0e0e0 1px;
  margin-bottom: 3rem;
`;

const Title = styled.h1`
  color: #553c9a;
  font-family: 'Poppins', sans-serif;
`;

const Link = styled(ReachRouterLink)`
  background-color: #e0e0e0;
  margin: 1rem;
  height: fit-content;
  color: #553c9a;
  border: 0;
  font-size: 1.2em;
  border-radius: 3px;
  padding: 8px 17px;
  cursor: pointer;
  text-decoration: none;
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;

  &:hover {
    background: #553c9a;
    color: #f4f4f4;
  }
`;

function Header() {
  return (
    <Wrapper>
      <Title>Math Magicians</Title>
      <nav>
        <Link to="/">Home</Link>
        <Link to="/calculator">Calculator</Link>
        <Link to="/quote">Quote</Link>
      </nav>
    </Wrapper>
  );
}
export default Header;
