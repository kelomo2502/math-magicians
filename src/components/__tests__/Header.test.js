import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Header';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
