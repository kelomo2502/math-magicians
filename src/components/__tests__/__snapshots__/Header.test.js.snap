// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bczRLJ ckUIWC"
>
  <h1
    className="sc-gsnTZi bXEcda"
  >
    Math Magicians
  </h1>
  <nav>
    <a
      className="sc-dkzDqf bXsskN"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="sc-dkzDqf bXsskN"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className="sc-dkzDqf bXsskN"
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </nav>
</div>
`;
